File: dotnet/Lexy/Lexy.Compiler/Language/Expressions/Functions/IntFunction.cs
using Lexy.Compiler.Parser;

namespace Lexy.Compiler.Language.Expressions.Functions
{
    public class IntFunction : SingleArgumentFunction
    {
        public const string Name = "INT";

        protected override string FunctionHelp => $"{Name} expects 1 arguments (Value)";

        protected override VariableType ArgumentType => PrimitiveType.Number;
        protected override VariableType ResultType => PrimitiveType.Number;

        private IntFunction(Expression valueExpression, SourceReference reference)
            : base(valueExpression, reference)
        {
        }

        public static BuiltInFunction Create(SourceReference reference, Expression expression) =>
            new IntFunction(expression, reference);
    }
}

File: typescipt/src/language/expressions/functions/intFunction.ts


export class IntFunction extends SingleArgumentFunction {
   public const string Name = `INT`;

   protected override string FunctionHelp => $`{Name} expects 1 argument (Value)`;

   protected override VariableType ArgumentType => PrimitiveType.number;
   protected override VariableType ResultType => PrimitiveType.number;

   constructor(valueExpression: Expression, reference: SourceReference)
     : base(valueExpression, reference) {
   }

   public static create(reference: SourceReference, expression: Expression): ExpressionFunction {
     return new IntFunction(expression, reference);
   }
}
